revisions-extended/includes/rest-revision-controller.php

```diff
@@ -314,6 +314,27 @@ protected function prepare_links( $post ) {
 		return $links;
 	}
 
+	/**
+	 * Gets an array of fields to be included on the response.
+	 *
+	 * @param WP_Post $post Post object.
+	 * @param string  $context String representing the context.
+	 *
+	 * @return WP_Response Response object
+	 */
+	public function filter_response_by_context( $post, $context ) {
+
+		if ( ! empty( $post['parent'] ) ) {
+			$parent = get_post( $post['parent'] );
+
+			if ( ! post_type_supports( $parent->post_type, 'excerpt' ) ) {
+				remove_post_type_support( $post['type'], 'excerpt' );
+			}
```

Comment:
```suggestion
			$parent_post_type = get_post_type( $post['parent'] );

			if ( post_type_supports( $parent_post_type, 'excerpt' ) ) {
				add_post_type_support( $post['type'], 'excerpt' );
			} else {
				remove_post_type_support( $post['type'], 'excerpt' );
			}
```

I like where you're going with this. I think the above change would cover a scenario where revisions from two different post types are prepared for the same request. That may not be possible in this rest controller as it exists right now, but if it does in the future, this would hopefully help us avoid a subtle bug.